### ISSUE TITLE
Error while packaging UE4.26.2

### ISSUE THREAD

**Issue by yexiuph** [getnamo/socketio-client-ue4#284]
![image](https://user-images.githubusercontent.com/42666658/131687882-48f2ada8-d047-41fa-b838-ba98233541b1.png)

**Comment by getnamo** [comment]
Plugin version?

**Comment by crafteke** [comment]
I think it's related to my issue #272 
I got it working perfectly on windows, on linux I've the same error.

**Comment by yexiuph** [comment]
> I think it's related to my issue #272 
> I got it working perfectly on windows, on linux I've the same error.
> 
> 

Probably same. Will be checking your thread later

**Comment by yexiuph** [comment]
> Plugin version?

Latest

**Comment by getnamo** [comment]
> > Plugin version?
> 
> Latest

Specific version?

**Comment by yexiuph** [comment]
> > > Plugin version?
> > 
> > 
> > Latest
> 
> Specific version?
v1.6.0 GitHub Version non Marketplace

**Comment by yexiuph** [comment]
![image](https://user-images.githubusercontent.com/42666658/132711866-be70b5a2-61eb-47ff-87ed-161565926cb2.png)


Tried moving the Plugin into Source but still no luck.

**Comment by yexiuph** [comment]
Quick Update:  Version 1.5.5 works fine with packaging - Please use it in the meantime.

**Comment by getnamo** [comment]
Yep v1.6.0 hasn't be fully tested on all platforms, 1.5.5 is the latest fully stable version.

**Comment by crafteke** [comment]
For me it gives me:
````
In file included from /home/hachpai/Documents/Unreal Projects/socketiotest/Plugins/socketio-client-ue4-1.5.5/Intermediate/Build/Linux/B4D820EA/UE4Editor/Development/SocketIOLib/Module.SocketIOLib.cpp:2:
In file included from /home/hachpai/Documents/Unreal Projects/socketiotest/Plugins/socketio-client-ue4-1.5.5/Source/SocketIOLib/Private/internal/sio_client_impl.cpp:28:
/home/hachpai/Documents/Unreal Projects/socketiotest/Plugins/socketio-client-ue4-1.5.5/Source/SocketIOLib/Private/internal/sio_client_impl.h:34:10: fatal error: 'websocketpp/client.hpp' file not found
#include <websocketpp/client.hpp>
         ^~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
````

**Comment by getnamo** [comment]
Clone with submodules see readme

**Comment by crafteke** [comment]
Sorry, I copy pasted the wrong output!
I git cloned with submodules and got :
````
./ue4build.sh 

Running Mono...

Fixing inconsistent case in filenames.
Setting up Mono
~/epic/UnrealEngine-4.26/Engine ~/Documents/Unreal Projects/socketiotest
Using 'git status' to determine working set for adaptive non-unity build (/home/hachpai/epic/UnrealEngine-4.26).
Creating makefile for UE4Editor (command line arguments changed)
@progress push 5%
@progress 'Generating code...' 0%
@progress push 5%
@progress 'Generating code...' 0%
@progress 'Generating code...' 67%
@progress 'Generating code...' 100%
@progress pop
@progress 'Generating code...' 33%
Parsing headers for UE4Editor
  Running UnrealHeaderTool "/home/hachpai/Documents/Unreal Projects/socketiotest/socketiotest.uproject" "/home/hachpai/Documents/Unreal Projects/socketiotest/Intermediate/Build/Linux/B4D820EA/UE4Editor/Development/UE4Editor.uhtmanifest" -LogCmds="loginit warning, logexit warning, logdatabase error" -Unattended -WarningsAsErrors -abslog="/home/hachpai/epic/UnrealEngine-4.26/Engine/Programs/UnrealBuildTool/Log_UHT.txt"
Reflection code generated for UE4Editor in 4.3778562 seconds
@progress 'Generating code...' 67%
@progress 'Generating code...' 100%
@progress pop
------- Build details --------
Using toolchain located at '/home/hachpai/epic/UnrealEngine-4.26/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v17_clang-10.0.1-centos7/x86_64-unknown-linux-gnu'.
Using clang (/home/hachpai/epic/UnrealEngine-4.26/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v17_clang-10.0.1-centos7/x86_64-unknown-linux-gnu/bin/clang++) version '10.0.1' (string), 10 (major), 0 (minor), 1 (patch)
Using bundled libc++ standard C++ library.
Using lld linker
Using llvm-ar : /home/hachpai/epic/UnrealEngine-4.26/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v17_clang-10.0.1-centos7/x86_64-unknown-linux-gnu/bin/llvm-ar
Using fast way to relink  circularly dependent libraries (no FixDeps).
------------------------------
Building UE4Editor...
Performing 15 actions (12 in parallel)
@progress 'Compiling C++ source code...' 0%
[3/15] Compile Module.SIOJson.cpp
[2/15] Compile Module.CoreUtility.gen.cpp
[1/15] Compile Module.CoreUtility.cpp
[6/15] Compile Module.SIOJEditorPlugin.cpp
[4/15] Compile Module.SIOJson.gen.cpp
[8/15] Compile Module.SocketIOClient.cpp
[9/15] Compile Module.SocketIOClient.gen.cpp
[7/15] Compile Module.SIOJEditorPlugin.gen.cpp
[5/15] Compile Module.SocketIOLib.cpp
@progress 'Compiling C++ source code...' 6%
@progress 'Compiling C++ source code...' 12%
@progress 'Compiling C++ source code...' 19%
@progress 'Compiling C++ source code...' 25%
@progress 'Compiling C++ source code...' 31%
@progress 'Compiling C++ source code...' 38%
[10/15] Link (lld) libUE4Editor-CoreUtility-0001.so
@progress 'Compiling C++ source code...' 44%
@progress 'Compiling C++ source code...' 50%
@progress 'Compiling C++ source code...' 56%
[11/15] Link (lld) libUE4Editor-SIOJson-0001.so
ld.lld: error: unable to find library -lUE4Editor-CoreUtility
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
@progress 'Compiling C++ source code...' 62%
@progress 'Compiling C++ source code...' 69%
@progress 'Compiling C++ source code...' 88%
[12/15] Link (lld) libUE4Editor-SocketIOLib-0001.so
@progress 'Compiling C++ source code...' 94%

````

**Comment by crafteke** [comment]
> Clone with submodules see readme

Any idea?
You could compile on linux?

**Comment by crafteke** [comment]
Ok I just  launched my editor and it built successfully!
I patched UE and modify the file  `Engine/Source/Developer/DesktopPlatform/Private/DesktopPlatformBase.cpp 
`  line 579, 
````
	//Arguments += " -Progress -NoEngineChanges -NoHotReloadFromIDE";
	Arguments += " -Progress";
````
Then i went back to root ue folder and did:
````
./Setup.sh
./GenerateProjectFiles.sh
make
````
And the editor could launch my project with the plugin. It appears correctly in BP components.

**Comment by ChrisH-001** [comment]
FYI: I've posted the fix I used for the SocketIO V1.60 #if errors (and another issue) over to https://github.com/getnamo/socketio-client-ue4/issues/272
